src/multivariate.jl

```diff
@@ -1,52 +1,82 @@
 
 A multivariate normal distribution whose covariance is diagonal. Compatible with Tracker.
 """
-struct TuringDiagNormal{Tm<:AbstractVector, Tσ<:AbstractVector} <: ContinuousMultivariateDistribution
+struct TuringDiagMvNormal{Tm<:AbstractVector, Tσ<:AbstractVector} <: ContinuousMultivariateDistribution
     m::Tm
     σ::Tσ
 end
 
-Distributions.dim(d::TuringDiagNormal) = length(d.m)
-function Distributions.rand(rng::Random.AbstractRNG, d::TuringDiagNormal)
-    return d.m .+ d.σ .* randn(rng, dim(d))
+Base.length(d::TuringDiagMvNormal) = length(d.m)
+Base.size(d::TuringDiagMvNormal) = (length(d), length(d))
+function Distributions.rand(rng::Random.AbstractRNG, d::TuringDiagMvNormal)
+    return d.m .+ d.σ .* randn(rng, length(d))
 end
+function Distributions.rand(rng::Random.AbstractRNG, d::TuringDiagMvNormal, n::Int)
+    return d.m .+ d.σ .* randn(rng, length(d), n)
+end
+
+struct TuringScalMvNormal{Tm<:AbstractVector, Tσ<:Real} <: ContinuousMultivariateDistribution
+    m::Tm
+    σ::Tσ
+end
+
+Base.length(d::TuringScalMvNormal) = length(d.m)
+Base.size(d::TuringScalMvNormal) = (length(d), length(d))
+function Distributions.rand(rng::Random.AbstractRNG, d::TuringScalMvNormal)
+    return d.m .+ d.σ .* randn(rng, length(d))
+end
+function Distributions.rand(rng::Random.AbstractRNG, d::TuringScalMvNormal, n::Int)
+    return d.m .+ d.σ .* randn(rng, length(d), n)
+end
+
 for T in (:AbstractVector, :AbstractMatrix)
-    @eval Distributions.logpdf(d::TuringDiagNormal, x::$T) = _logpdf(d, x)
-    @eval Distributions.logpdf(d::TuringMvNormal, x::$T) = _logpdf(d, x)
+    @eval Distributions.logpdf(d::TuringScalMvNormal, x::$T) = _logpdf(d, x)
+    @eval Distributions.logpdf(d::TuringDiagMvNormal, x::$T) = _logpdf(d, x)
+    @eval Distributions.logpdf(d::TuringDenseMvNormal, x::$T) = _logpdf(d, x)
 end
 
-function _logpdf(d::TuringDiagNormal, x::AbstractVector)
-    return -(dim(d) * log(2π) + 2 * sum(log.(d.σ)) + sum(abs2, (x .- d.m) ./ d.σ)) / 2
+function _logpdf(d::TuringScalMvNormal, x::AbstractVector)
+    return -(length(x) * log(2π * abs2(d.σ)) + sum(abs2, (x .- d.m) ./ d.σ)) / 2
 end
-function _logpdf(d::TuringDiagNormal, x::AbstractMatrix)
-    return -(dim(d) * log(2π) .+ 2 * sum(log.(d.σ)) .+ sum(abs2, (x .- d.m) ./ d.σ, dims=1)') ./ 2
+function _logpdf(d::TuringScalMvNormal, x::AbstractMatrix)
+    return -(size(x, 2) * log(2π) .+ 2 * sum(log(d.σ)) .+ sum(abs2, (x .- d.m) ./ d.σ, dims=1)') ./ 2
 end
-function _logpdf(d::TuringMvNormal, x::AbstractVector)
-    return -(dim(d) * log(2π) + logdet(d.C) + sum(abs2, zygote_ldiv(d.C.U', x .- d.m))) / 2
+
+function _logpdf(d::TuringDiagMvNormal, x::AbstractVector)
+    return -(length(x) * log(2π) + 2 * sum(log.(d.σ)) + sum(abs2, (x .- d.m) ./ d.σ)) / 2
+end
+function _logpdf(d::TuringDiagMvNormal, x::AbstractMatrix)
+    return -(size(x, 2) * log(2π) .+ 2 * sum(log.(d.σ)) .+ sum(abs2, (x .- d.m) ./ d.σ, dims=1)') ./ 2
```

Comment:
```suggestion
    return -(size(x, 2) * log(2π) .+ 2 * sum(log, d.σ) .+ vec(sum(abs2, (x .- d.m) ./ d.σ, dims=1))) ./ 2
```

Reply:
The broadcasting versions is actually a little faster with `TrackedVector` but the opposite is true for normal vectors.